src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -78,6 +78,62 @@ public static RefKind GetRefKind(this BoundExpression node)
             }
         }
 
+        /// <summary>
+        /// Returns the RefKind of an expression
+        /// if the expression represents a symbol, the RefKind of the symbol is returned
+        /// if the expression is a "by-ref" expression like conditional operator or InlineArrayAccess the RefKind is calculated and returned
+        /// otherwise this function returns RefKind.None 
+        /// </summary>
+        public static RefKind GetRefKindEx(this BoundExpression node)
+        {
+            switch (node.Kind)
+            {
+                case BoundKind.ConditionalOperator:
+                    {
+                        var conditionalOperator = (BoundConditionalOperator)node;
+                        if (!conditionalOperator.IsRef)
+                        {
+                            return RefKind.None;
+                        }
+                        //the ref kind of the entire expression is ref readonly if the ref kind of the consequence
+                        //or the alternative is ref readonly otherwise it just by ref (not readonly)
+                        //
+                        //without this check we loose the readonly specifier
+                        //See Binder.CheckValueKind
+                        if (tryGetReadOnlyRefKind(conditionalOperator.Consequence) is { } readOnlyRefKindOfConsequence)
+                        {
+                            return readOnlyRefKindOfConsequence;
+                        }
+                        if (tryGetReadOnlyRefKind(conditionalOperator.Alternative) is { } readOnlyRefKindOfAlternative)
+                        {
+                            return readOnlyRefKindOfAlternative;
+                        }
+                        return RefKind.Ref;
+
+                        static RefKind? tryGetReadOnlyRefKind(BoundExpression expression)
+                        {
+                            switch (expression.GetRefKindEx())
+                            {
+                                case RefKind.Ref:
+                                case RefKind.Out:
+                                    return null;
+                                case RefKind.RefReadOnlyParameter:
+                                case RefKind.RefReadOnly: //same as RefKind.In
+                                case RefKindExtensions.StrictIn:
```

Comment:
In some cases ```RefKindExtensions.StrictIn``` is actually handled differently by comparison to RefKind.RefReadOnly. I think we should return ```RefKindExtensions.StrictIn``` for it and, in case of an ambiguity between ```RefKindExtensions.StrictIn``` and ```RefKind.RefReadOnly``` across branches, we should prefer ```RefKindExtensions.StrictIn```

Reply:
Probably always using ```RefKindExtensions.StrictIn``` instead of ```RefKind.RefReadOnly``` is a better alternative.